doc/dev/mfx_trace2-dev.md

```diff
@@ -0,0 +1,75 @@
+# MediaSDK tracing (mfx_trace2). Developer documentation
+
+## How to use mfx_trace2?
+
+See [mfx_trace2 user documentation](./mfx_trace2.md).
+
+## Overview
+
+Main trace system object is presented by global instance of `mfx::Trace` class (instance is called `_mfx_trace`).
+Its constructor (`mfx::Trace::Trace()`) reads configuration file (on Linux) or registry values (on Windows), initializes all enabled backends.
+
+User is not supposed to touch internals of `mfx::Trace` class except `mfx::Trace::Scope` which is public interface for user to enable scoped tracing. Scope saves all events in std::vector `_mfx_trace.events` where all backends can find them using `mfx::Trace::pushEvent` function.
+
+Event saves the following information:
+- SourceLocation (`mfx::SourceLocation` class)
+  It contains file name, line and function name where event was called.
+- name
+- category (it can be specified by users in `mfx::Trace::Scope` constructor)
+- event type
+  - BEGIN (scope constructor)
+  - END (scope destructor)
+  - INFO (`event` method call of `mfx::Trace::Scope` is called)
+  - other (other types are reserved for creating custom backend-specific events, for example, OTHER1 and OTHER2 are used in Chrome backend for creating link events))
+- id (it is the same among one BEGIN-INFO-END group of events)
+- parentIndex (in events of type BEGIN it is always zero, if it is INFO or END type of event then it contains index of corresponding BEGIN event in `_mfx_trace.events` array)
+- timestamp (in microseconds)
+- threadId
+- description
+- map (contains key-value pairs of information that is added using `mfx::Trace::Scope::event(key, value)` method)
```

Comment:
map of key-value pairs... (no need in braces)

Reply:
Done